internal/alb/sg/instance_attachment.go

```diff
@@ -0,0 +1,199 @@
+package sg
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/alb/tg"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/aws/albec2"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/ingress/controller/store"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/pkg/util/log"
+)
+
+// InstanceAttachment represents the attachment of securityGroups to instance
+type InstanceAttachment struct {
+	GroupID string
+	Targets tg.TargetGroups
+}
+
+// InstanceAttachementController manages InstanceAttachment
+type InstanceAttachementController interface {
+	Reconcile(*InstanceAttachment) error
```

Comment:
Could you add some docs on what does Reconcile do? In general, all the public APIs should be documented (golint will complain once we add it). It is not urgent now, but could help a lot for other contributor to understand the code.

Reply:
added docs on these methods